service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -372,6 +403,16 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
             }
         }
 
+        if (endPoint.isIncludeMetaInfo()) {
+            if (endPoint.getCommandName() != null && !endPoint.getCommandName().isEmpty()) {
+                block.decl(jCodeModel.ref("String"), COMMAND_NAME_VAR_NAME, JExpr.lit(endPoint.getCommandName()));
+            } else {
+                block.decl(jCodeModel.ref("String"), COMMAND_NAME_VAR_NAME, JExpr.invoke("getCommandName"));
+            }
+            JInvocation invocation = jCodeModel.ref("String").staticInvoke("valueOf").arg(JExpr.ref(META_INFO_PARAMETER).invoke("get").arg("commandName"));
+            block._if(JExpr.ref(META_INFO_PARAMETER).invoke("containsKey").arg("commandName"))._then().assign(JExpr.ref(COMMAND_NAME_VAR_NAME), invocation);
```

Comment:
.get and checking for null is better than querying the map twice (once for contains and once for get)